### PR TITLE
update samples with dependency requirement in the component image build

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/378)

### CODE REVIEW

File: samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb [hunk 1]
```diff
@@ -421,9 +487,75 @@
...
+   ],
+   "source": [
+    "from kfp import compiler\n",
     "\n",
-    "Now that we've tested the function locally, we want to build a component that can run as a step in the pipeline. First we need to build a base docker container image. We need TensorFlow and google-api-python-client packages."
+    "# The return value \"DeployerOp\" represents a step that can be used directly in a pipeline function\n",
+    "DeployerOp = compiler.build_python_component(\n",
+    "    component_func=deploy_model,\n",
+    "    staging_gcs_path=OUTPUT_DIR,\n",
+    "    dependency=[kfp.compiler.VersionedDependency(name='google-api-python-client', version='1.7.0')],\n",
```
Comment (qimingj) [inline samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: Not related to this change. Can we do:

dependency=['google-api-python-client==1.7.0', 'numpy>=1.2', 'tensorflow']

instead if having a "VersionedDependency" wrapper?

Comment (qimingj) [reply samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: I understand. It should be a separate change.

I am not worried about user not knowing detailed format. It is a well known format for Python users, and there are good docs to point them to (setup.py, requirements.txt, etc). For us, we just need to take the strings, maybe do some simple validations, and then pass them to Python's world. I feel that reduces the surface area. To me, it is always nice to align to a certain accepted convention. Let's also chat offline about this.

Comment (gaoning777) [reply samples/notebooks/KubeFlow Pipeline Using TFX OSS Components.ipynb]: Sure. Let's sync offline.

**Pull request merged**